pennylane/wires.py

```diff
@@ -493,6 +493,186 @@ def unique_wires(list_of_wires):
 
         return Wires(tuple(unique), _override=True)
 
+    def union(self, other):
+        """Return the union of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1.union(wires2)
+        Wires([1, 2, 3, 4, 5])
+        """
+        if not isinstance(other, Wires):
+            raise TypeError(f"Can only do the union of Wires with Wires. Got {type(other)}.")
```

Comment:
Just putting a note here that the error message for each set combination method should reflect that we allow for `other` to be a `list`